xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/layoutvars.vm

```diff
@@ -119,6 +119,9 @@
  #end
 #end
 
+## Check that the doc velocity variable is defined: it might not be the case when using some resource handlers.
+#set ($docIsDefined = ("$!doc" != ''))
```

Comment:
Not sure why you compare it to an empty String, a simple `#if ($doc)` (which check if it's not null) would be enough IMO. Feels more accurate than calling XWikiDocument#toString().

Reply:
I'm using this pattern when I want a one-liner, it avoids
```
#if ($doc)
  #set ($docIsDefined = true)
#else 
  #set ($docIsDefined = false)
#end 
```
but if you have a better way for that kind of stuff I'm taking it.

Reply:
hmm maybe I could use `$doc != $NULL`?

Reply:
That should work, another possibility is `#set ($docIsDefined = !!$doc)` (putting a `!` in front of a variable in a set cause the set to set as value the boolean result of the if and then you invert it again with another `!`).

Reply:
Hmm I prefer `$doc != $NULL` it's easier to read IMO.